lib/utils/which.rb

```diff
@@ -0,0 +1,16 @@
+# encoding: utf-8
+
+# Cross-platform way of finding an executable in the $PATH.
+#
+#   which('ruby') #=> /usr/bin/ruby
+def which(cmd)
```

Comment:
Also, this seems like a great candidate for Memoize. Its return value should be the same for any given input, and running it again is potentially expensive.